beacon-chain/core/electra/registry_updates.go

```diff
@@ -0,0 +1,66 @@
+package electra
+
+import (
+	"context"
+
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/time"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/validators"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+)
+
+// ProcessRegistryUpdates rotates validators in and out of active pool.
+// the amount to rotate is determined churn limit.
+//
+// Spec pseudocode definition:
+//
+//		def process_registry_updates(state: BeaconState) -> None:
+//		    # Process activation eligibility and ejections
+//		    for index, validator in enumerate(state.validators):
+//		        if is_eligible_for_activation_queue(validator):
+//		            validator.activation_eligibility_epoch = get_current_epoch(state) + 1
+//
+//		        if (
+//		            is_active_validator(validator, get_current_epoch(state))
+//		            and validator.effective_balance <= EJECTION_BALANCE
+//		        ):
+//		            initiate_validator_exit(state, ValidatorIndex(index))
+//
+//	         # Activate all eligible validators
+//	         activation_epoch = compute_activation_exit_epoch(get_current_epoch(state))
+//	         for validator in state.validators:
+//	             if is_eligible_for_activation(state, validator):
+//	                 validator.activation_epoch = activation_epoch
+func ProcessRegistryUpdates(ctx context.Context, state state.BeaconState) (state.BeaconState, error) {
```

Comment:
you can add a span for this like your other PRs

Reply:
I dont know that it is necessary yet. Perhaps other spans should be removed unless it is considered a hotspot in processing